### DISCUSSION TITLE
model viewer vs. 3D Builder - how to achieve 3D Builder rendering?

### DISCUSSION THREAD

**Discussion by phovorka** [google/model-viewer#3335] [Q&A]
Hi i noticed quite different result with this two "Viewers" . Is there any way how to render GLB file in model-viewer more smoothly? like in 3D builder...just not see polygons
see pictures->
model-viewer
![modelV](https://user-images.githubusercontent.com/67637867/161227581-701e79ae-96ea-4989-ace9-b307a20d189b.PNG)

3D builder
![3dbuilder](https://user-images.githubusercontent.com/67637867/161227613-23f0040d-6f00-4071-bbb2-9a92efb3ffb3.PNG)
 
Thanks Emmett :-)

**Comment by elalish** [comment]
That's funny! I worked on 3D Builder years ago when it was new and I was at Microsoft. Anyway, technically 3D Builder is not rendering glTF correctly: the spec says it should be rendered faceted if there are no vertex normals in the file, and smooth if there are. If you re-export it with normals it should look nice and smooth in model-viewer too.

**Comment by phovorka** [comment]
Notice:Model without normals -> less datasize.
Rendering smooth without normals can be very nice feature not a bug:-) 

Is it possible to render model smoothly without normals data?

Have a nice day

**Reply by elalish** [reply]
No, it's part of the glTF spec and it can be hard to calculate good normals at mesh edges, like where two materials meet. If you're worried about file size, I'd recommend looking at DRACO compression, which can also compress the normal and uv channels. Generally the file size of a well-authored model is 80-90% texture images.
